libs/model/src/main/java/de/cyface/model/MetaData.kt

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2020-2025 Cyface GmbH
+ *
+ * This file is part of the Serialization.
+ *
+ * The Serialization is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Serialization is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the Serialization. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.cyface.model
+
+import java.io.Serializable
+import java.util.Date
+import java.util.Objects
+import java.util.UUID
+
+/**
+ * The context of a deserialized `Measurement`.
+ */
+class MetaData : Serializable {
```

Comment:
Maybe create a data class here. In that case you can avoid having to implement `toString`, `equals` and `hashCode`.

Reply:
This is not possible:
- we need a no-arg constructor
- thus we cannot call `init { require(version in SUPPORTED_VERSIONS...`

I ended up doing the following:

```
@NoArg # kotlin no-arg plugin
class MetaData private constructor(...) {
   fun validate() { require version to be ... }
   implement toString etc manually as the data class would: "Even if you make the primary constructor private, Kotlin still generates a public copy(...) method that bypasses your validation."
   companion object {
      // factory to ensure validate is called
      fun create(...) = MetaData(...).apply { validate() }
```